ui/src/components/DeHeader.js

```diff
@@ -68,70 +48,97 @@ const styles = {
-  state = {
-    snackbarOpen: true
-  };
-
   render() {
-    const { classes } = this.props;
+    const { classes, invalidFilterFacets, invalidExtraFacets } = this.props;
 
-    const snackbarKey = "hasShownSnackbarv2";
-    let snackbar = null;
-    if (localStorage.getItem(snackbarKey) === null) {
-      localStorage.setItem(snackbarKey, "true");
-      snackbar = (
+    const howToUseKey = "hasShownSnackbarv2";
+    let howToUseSnackbar = null;
+    if (localStorage.getItem(howToUseKey) === null) {
+      localStorage.setItem(howToUseKey, "true");
+      howToUseSnackbar = (
         <Snackbar
-          className={classes.snackbarRoot}
-          anchorOrigin={{
-            vertical: "top",
-            horizontal: "right"
-          }}
-          open={this.state.snackbarOpen}
-          onClose={this.handleSnackbarClose}
-          TransitionComponent={TransitionLeft}
-        >
-          <SnackbarContent
-            classes={{
-              message: classes.snackbarContentMessage,
-              root: classes.snackbarContentRoot
-            }}
-            message={
-              <>
-                <div>
-                  <b>How to use Data Explorer</b>
-                </div>
-                <br />
-                <div>Click on a bar to select it</div>
-                <div>
-                  <i>Hint:&nbsp;</i> Clicking on whitespace also works
-                </div>
-              </>
-            }
-            action={[
-              <IconButton
-                key="close"
-                aria-label="Close"
-                color="inherit"
-                onClick={this.handleSnackbarClose}
-                className={classes.snackbarCloseButton}
-              >
-                <CloseIcon className={classes.snackbarCloseIcon} />
-              </IconButton>
-            ]}
-          />
-        </Snackbar>
+          message={
+            <>
+              <div>
+                <b>How to use Data Explorer</b>
+              </div>
+              <br />
+              <div>Click on a bar to select it</div>
+              <div>
+                <i>Hint:&nbsp;</i> Clicking on whitespace also works
+              </div>
+            </>
+          }
+        />
       );
     }
 
+    const invalidFilterFacetsSnackbar =
+      invalidFilterFacets && invalidFilterFacets.length
+        ? createInvalidFacetsSnackbar(
+            classes,
+            invalidFilterFacets,
+            <div>Try recreating your cohort and saving to Terra again.</div>
+          )
+        : null;
+
+    const invalidExtraFacetsSnackbar =
+      invalidExtraFacets && invalidExtraFacets.length
+        ? createInvalidFacetsSnackbar(
+            classes,
+            invalidExtraFacets,
+            <div>
+              Try searching for
+              {invalidFilterFacets.length === 1
+                ? " this facet "
+                : " these facets "}
+              and saving cohort to Terra again.
```

Comment:
Does "saving *the* cohort to Terra again" sound better?

Reply:
Done